### DISCUSSION TITLE
Please display 24-hour time format (instead of AM/PM) when relevant

### DISCUSSION THREAD

**Discussion by keunes** [pretalx/pretalx#1192] [Ideas] [answered]
Hello!

Via [JoinUp](https://joinup.ec.europa.eu/collection/open-source-observatory-osor/event/online-event-matomo-camp) I just came across MatomoCamp's pretalx instance: https://schedule.matomocamp.org/matomocamp-2021/schedule/

As I'm based in Europe, I'm used to the 24-hour time format. Pretalx, however, seems to use AM/PM everywhere in the interface. It would be great if the 24-format is used in regions wherever that is logic. I imagine this could be done based on the browser's locale?

**Comment by rixx** [comment, marked as answer]
As you suggested, time formats are determined not by geographical location, but by your locale – though not your browser locale, it's by your pretalx language. For example, when using pretalx in German or French, times are shown in a 24-hour format.

Going off [this map](https://en.wikipedia.org/wiki/Date_and_time_representation_by_country#/media/File:12_24_Hours_World_Map.svg), using am/pm format seems to be a decent compromise for English-speaking countries, since it's in use or understood everywhere where English is used. I'm not sure where else we would add a manual toggle for this, and as you can see, it's not a simple "is_in_europa" kind of differentiation.

**Reply by keunes** [reply]
As English is used in such a wide range of countries (not only in English-speaking countries), a toggle at admin/server level (for not-logged in users, e.g. potential conference visitors) and at user level would be helpful indeed.

(We can't reliable use the map you linked to determine the best default, because many international events in Europe and in other places that use 24-h formatting are held in English.)

**Reply by rixx** [reply]
Yes. That is both true and tracked at #1269, keeping this discussion open would not serve any purpose.

**Comment by ericfreyss** [comment]
I have run into the same problem and written a solution that I have used:

# Enabling 24h Time Format and English (UK) in Pretalx

This guide explains how to force a 24-hour time format on a Pretalx instance by adding the `en-gb` (English UK) locale, while preserving all other installed languages.

This method uses a custom Python settings file to avoid modifying the Pretalx source code directly, ensuring your changes persist through future updates.

## 1. Create the Custom Settings File

Create a file named `custom_settings.py` in your Pretalx data directory (e.g., `/var/pretalx/`).

```python
from pretalx.settings import *
from django.utils.translation import gettext_noop

# 1. Inject required metadata for Pretalx
LANGUAGES_INFORMATION.update({
    'en-gb': {
        'code': 'en-gb',
        'natural_name': 'English (UK)',
        'official': 'English (UK)',
        'rtl': False,
        'visible': True,
        'percentage': 100,
    }
})

# 2. Add English (UK) to the existing language list without overwriting others
if not any(lang[0] == 'en-gb' for lang in LANGUAGES):
    LANGUAGES = list(LANGUAGES) + [('en-gb', gettext_noop('English (UK)'))]

```

Set the correct ownership for the file:
```bash
sudo chown pretalx:pretalx /var/pretalx/custom_settings.py
```

## 2. Systemd Configuration

To tell Pretalx to use this file instead of the default settings, modify the service unit using an override file.

1.  **Open the override editor:**
    ```bash
    sudo systemctl edit pretalx-web.service
    ```
2.  **Add the following environment variables:**
    ```ini
    [Service]
    Environment="PYTHONPATH=/var/pretalx"
    Environment="DJANGO_SETTINGS_MODULE=custom_settings"
    ```

## 3. Apply Changes

Reload the systemd configuration and restart the Pretalx web service:
```bash
sudo systemctl daemon-reload
sudo systemctl restart pretalx-web
```

## 4. UI Configuration

Once the service has restarted:

1.  Log in to your event's administration interface.
2.  Navigate to **Settings** > **General** > **Internationalization**.
3.  Check **English (UK)** in the available languages list.
4.  Set **English (UK)** as the **Default language**.
5.  Save your changes.

**Reply by rixx** [reply]
FWIW the real fix in pretalx is in handling languages and locales separately; ideally both on an organiser level and on a user/attendee level. I'm happy to review a PR to that effect any time.

Side note: Could you explain why you chose to use `gettext_noop('English (UK)'` ?

**Reply by ericfreyss** [reply]
I am not sure I am agile enough to develop that PR myself but yes I agree, the solution is only a workaround.
I used gettext_noop to include it in the Django translation workflow
